modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/upload/DDMUserPersonalFolderUploadFileEntryHandler.java

```diff
@@ -86,20 +103,31 @@ public FileEntry upload(UploadPortletRequest uploadPortletRequest)
 		}
 
 		try (InputStream inputStream = uploadPortletRequest.getFileAsStream(
-				_PARAMETER_NAME)) {
-
-			long repositoryId = ParamUtil.getLong(
-				uploadPortletRequest, "repositoryId");
+				parameterName)) {
 
-			String uniqueFileName = _uniqueFileNameProvider.provide(
-				fileName,
-				curFileName -> _exists(repositoryId, folderId, curFileName));
+			if (fileEntryId == 0) {
+				long repositoryId = ParamUtil.getLong(
+					uploadPortletRequest, "repositoryId");
+
+				String uniqueFileName = _uniqueFileNameProvider.provide(
+					fileName,
+					curFileName -> _exists(
+						repositoryId, folderId, curFileName));
+
+				return _dlAppService.addFileEntry(
+					null, repositoryId, folderId, uniqueFileName,
+					uploadPortletRequest.getContentType(parameterName),
+					uniqueFileName, uniqueFileName,
+					_getDescription(uploadPortletRequest), StringPool.BLANK,
+					inputStream, size, null, null,
+					_getServiceContext(uploadPortletRequest));
+			}
 
-			return _dlAppService.addFileEntry(
-				null, repositoryId, folderId, uniqueFileName,
-				uploadPortletRequest.getContentType(_PARAMETER_NAME),
-				uniqueFileName, uniqueFileName,
-				_getDescription(uploadPortletRequest), StringPool.BLANK,
+			return _dlAppService.updateFileEntry(
+				fileEntryId, fileName,
+				uploadPortletRequest.getContentType(parameterName), fileName,
+				fileName, _getDescription(uploadPortletRequest),
+				StringPool.BLANK, DLVersionNumberIncrease.AUTOMATIC,
 				inputStream, size, null, null,
 				_getServiceContext(uploadPortletRequest));
```

Comment:
@TaiPhamLR @locpham97 I tested a similar scenario in the Web Content with the Image field and I noticed that we are not updating the existing image but creating a new one. Could you confirm with the team the expected behavior in this case?

You can see in the following video that two images are added after editing the original image:

[Gravação de tela de 09-03-2023 11:19:18.webm](https://user-images.githubusercontent.com/34186632/224053643-e49b642c-0770-455b-a72d-0907a0601ed5.webm)

Reply:
Hi @carolmariaabb ,
I will ask the reporter about this issue.
Thanks,
Loc

Reply:
Hi @carolmariaabb ,
I updated the code and remove the update function.
Thanks,
Loc